substrate/frame/support/src/lib.rs

```diff
@@ -1019,1222 +909,358 @@ pub mod pallet_prelude {
-///     * [`ValidateUnsigned`](`pallet_prelude::ValidateUnsigned`) was moved inside the macro
-///       under
-/// 	  [`#[pallet::validate_unsigned)]`](#validate-unsigned-palletvalidate_unsigned-optional)
-/// 	  if it exists
-///     * [`ProvideInherent`](`pallet_prelude::ProvideInherent`) was moved inside the macro
-///       under [`#[pallet::inherent)]`](#inherent-palletinherent-optional) if it exists
-///     * `on_initialize` / `on_finalize` / `on_runtime_upgrade` / `offchain_worker` were moved
-///       to the `Hooks` implementation
-///     * storages with `config(..)` were converted to `GenesisConfig` field, and their default
-///       is `= $expr;` if the storage has a default value
-///     * storages with `build($expr)` or `config(..)` were built in `GenesisBuild::build`
-///     * `add_extra_genesis` fields were converted to `GenesisConfig` field with their correct
-///       default if specified
-///     * `add_extra_genesis` build was written into `GenesisBuild::build`
-/// * storage items defined with [`pallet`] use the name of the pallet provided by
-///   [`traits::PalletInfo::name`] as `pallet_prefix` (in `decl_storage`, storage items used
-///   the `pallet_prefix` given as input of `decl_storage` with the syntax `as Example`). Thus
-///   a runtime using the pallet must be careful with this change. To handle this change:
-///     * either ensure that the name of the pallet given to `construct_runtime!` is the same
-///       as the name the pallet was giving to `decl_storage`,
-///     * or do a storage migration from the old prefix used to the new prefix used.
-///
-/// NOTE: The prefixes used by storage items are in metadata. Thus, ensuring the metadata
-/// hasn't changed ensures that the `pallet_prefix`s used by the storage items haven't changed.
-///
-/// # Notes when macro fails to show proper error message spans:
-///
-/// Rustc loses span for some macro input. Some tips to fix it:
-/// * do not use inner attribute:
-/// 	```ignore
-/// 	#[pallet]
-/// 	pub mod pallet {
-/// 		//! This inner attribute will make span fail
-/// 		..
-/// 	}
-/// 	```
-/// * use the newest nightly possible.
+/// This approach is beneficial when you use the `include_str` macro at the beginning of
+/// the file and want that documentation to extend to the runtime metadata, without
+/// reiterating the documentation on the pallet module itself.
 pub use frame_support_procedural::pallet;
 
 /// Contains macro stubs for all of the pallet:: macros
 pub mod pallet_macros {
 	pub use frame_support_procedural::{
-		composite_enum, config, disable_frame_system_supertrait_check, error, event,
-		extra_constants, feeless_if, generate_deposit, generate_store, getter, hooks,
-		import_section, inherent, no_default, no_default_bounds, origin, pallet_section,
-		storage_prefix, storage_version, type_value, unbounded, validate_unsigned, weight,
-		whitelist_storage,
+		composite_enum, config, disable_frame_system_supertrait_check, extra_constants,
```

Comment:
based on the title of the PR, I thought it had already documented all of these, but it seems not? is it WIP?

Reply:
They are all documented, the ones you see here have their docs written where they are defined in the `frame-support-procedural` crate. 

I originally had only moved docs here when I felt the benefit of deep linking to `frame` crates outweighed the cost of not having the docs inline in rust-analyzer. It kind of sucks when you hover and see the "Rust Analyzer Users: Please see docs at XYZ" instead of the docs themselves.

But I'm happy to move them all here if you prefer, which will also allow us to make all the docs compile.

Reply:
I suppose if we really really want to be pedantic, the ideal solution is this: 

- the main source of truth is in `frame-support`. 
- the secondary source of truth is `frame-support-procedural`
- either with a script or manually, in the latter, we make the latter be aidentical to the former, except 1) all intra-doc links like [`Foo`] change to `Foo` and 2) all code blocks receive an `ignore` tag.